File: Lab 2/Новикова А.С/src/stack.cpp
#include "stack.h"

Stack::Stack(int elemAmount)
{
	size = elemAmount;
	data = new Run[size];
	top = -1;
}
Stack::~Stack()
{
	delete[] data;
}

Run Stack::peak()
{
	return data[top];
}

void Stack::pop()
{
	if (isEmpty())
		std::cout << "The stack is empty! Nothing to pop!";

	else
		top--;

}

void Stack::push(Run value)
{
	if (isFull())
	{
		pushBack(value);
		top++;
	}

	else
		data[++top] = value;
}

bool Stack::isEmpty()
{
	return (top < 0);
}

bool Stack::isFull()
{
	return (top == size - 1);
}

int Stack::getSize()
{
	return top;
}

void Stack::pushBack(Run value)
{
	Run* temp = new Run[size + 1];
	for (int i = 0; i < size; i++)
		temp[i] = data[i];

	temp[size++] = value;

	delete[] * &data;
	data = temp;
}

File: Course work/Stoyanov/src/structures/Stack.py
from .LinkedList import LinkedList

class Stack():
    def __init__(self) -> None:
        self.linked_list: LinkedList =  LinkedList()
    
        
    def __len__(self) -> int:
        return self.size()
    
    
    def size(self) -> int:
        return len(self.linked_list)
    
    
    def is_empty(self) -> bool:
        return True if len(self.linked_list) == 0 else False
    
    
    def top(self) -> any:
        return self.linked_list.head.element
    
    
    def push(self, element: any) -> any:
        return self.linked_list.insert(0, element)
    
    
    def pop(self) -> any:
        return self.linked_list.delete(0)  

    def __str__(self) -> str:
        return str(self.linked_list)
